controllers/argocd/configmap.go

```diff
@@ -355,18 +357,26 @@ func (r *ReconcileArgoCD) reconcileArgoConfigMap(cr *argoprojv1a1.ArgoCD) error
 	cm.Data[common.ArgoCDKeyServerURL] = r.getArgoServerURI(cr)
 	cm.Data[common.ArgoCDKeyUsersAnonymousEnabled] = fmt.Sprint(cr.Spec.UsersAnonymousEnabled)
 
-	if !isDexDisabled() {
+	// create dex config if dex is enabled either through DISABLE_DEX or through `.spec.sso`
+
+	// make sure old workloads not using DISABLE_DEX don't slip through here because their .spec.sso is nil
+	if !isDexDisabled() && isDisableDexSet && (cr.Spec.SSO == nil || cr.Spec.SSO.Provider != v1alpha1.SSOProviderTypeDex) ||
```

Comment:
> If DISABLE_DEX=false and cr.Spec.SSO.Provider="dex" then this logic will skip configuring dex.
Is this a problem? I think we should be encouraging users to use one or the other

The `useDex` function sounds like a great idea, working on it now

Reply:
Not sure I follow.  For new users, with `DISABLE_DEX` not set, and no SSO configured (`.spec.sso = nil`), and `.spec.dex` not being set either, `useDex()` should return `false`

Reply:
> > If DISABLE_DEX=false and cr.Spec.SSO.Provider="dex" then this logic will skip configuring dex.
> 
> Is this a problem? I think we should be encouraging users to use one or the other

It may be a corner case, but I think users would find it very surprising that dex didn't get configured in this case.